java/src/org/openqa/selenium/remote/service/DriverCommandExecutor.java

```diff
@@ -77,7 +78,12 @@ public DriverCommandExecutor(DriverService service) {
    */
   protected DriverCommandExecutor(
       DriverService service, Map<String, CommandInfo> additionalCommands) {
-    super(additionalCommands, service.getUrl());
+    this(service, additionalCommands, ClientConfig.defaultConfig());
+  }
+
+  protected DriverCommandExecutor(
+    DriverService service, Map<String, CommandInfo> additionalCommands, ClientConfig clientConfig) {
+    super(additionalCommands, service.getUrl(), clientConfig);
```

Comment:
We would need to add an edit check that ensures that the `ClientConfig.baseUri` is set and if its not set, we would need to read the value from `DriverService` and then set it.